examples/chart/teleport-agent/values.yaml

```diff
@@ -0,0 +1,36 @@
+##################################################
+# Values that must always be provided by the user.
+##################################################
+
+# Join token for the cluster.
+authToken:
```

Comment:
Any reason you decided to use flat values structure, instead of organizing them in individual sections mirroring Teleport's config, e.g. auth, proxy, kube, etc. Would probably be easier to extend later with other services (app, db).

Reply:
This makes the CLI usage slightly more awkward:
```
$ helm install teleport-agent . \
  --namespace teleport \
  --set teleport.authServers=$PROXY_ENDPOINT \
  --set teleport.authToken=$JOIN_TOKEN \
  --set kubernetes_service.kubeClusterName=$KUBERNETES_CLUSTER_NAME
```
vs
```
$ helm install teleport-agent . \
  --namespace teleport \
  --set proxyAddr=$PROXY_ENDPOINT \
  --set authToken=$JOIN_TOKEN \
  --set kubeClusterName=$KUBERNETES_CLUSTER_NAME
```

My goal here is to make this chart single-purpose (register a k8s cluster in teleport) and as simple as possible.
We have https://github.com/gravitational/teleport/tree/master/examples/chart/teleport which is more suited to be the extensible universal chart.

Reply:
I think as the chart has one lone purpose it's fine to leave it as-is.

Reply:
Sounds good.